build/webvr-polyfill.js

```diff
@@ -1186,6 +1190,36 @@ VRDisplay.prototype.wrapForFullscreen = function(element) {
   parent.insertBefore(this.fullscreenWrapper_, this.fullscreenElement_);
   parent.removeChild(this.fullscreenElement_);
   this.fullscreenWrapper_.insertBefore(this.fullscreenElement_, this.fullscreenWrapper_.firstChild);
+  this.fullscreenElementCachedStyle_ = this.fullscreenElement_.getAttribute('style');
+
+  var self = this;
+  function applyFullscreenElementStyle() {
+    if (!self.fullscreenElement_)
+      return;
+
+    var cssProperties = [
+      'position: absolute',
+      'top: 0',
+      'left: 0',
+      'width: ' + Math.max(screen.width, screen.height) + 'px',
+      'height: ' + Math.min(screen.height, screen.width) + 'px',
+      'border: 0',
+      'margin: 0',
+      'padding: 0',
+    ];
+    self.fullscreenElement_.setAttribute('style', cssProperties.join('; ') + ';');
+  }
+
+  if (Util.isIOS()) {
+    // "To the last I grapple with thee;
+    //  from hell's heart I stab at thee;
+    //  for hate's sake I spit my last breath at thee."
+    // -- Moby Dick, by Herman Melville
+    this.fullscreenElement_.setAttribute('style', 'width: ' + (Math.max(screen.width, screen.height) - 1) + 'px;');
+    setTimeout(applyFullscreenElementStyle, 1000);
```

Comment:
@cvan , a few guesses from my experience battling this:

 - iOS 9.3+ has a scrolling bug that results in the canvas getting these wrong dimensions. zooming (css: zoom) to 1.1 and back to 1 fixed it for us.
 - iOS 7+ sometimes reports the wrong dimensions in some cases. (e.g. when the browser chrome is on screen)
 - iOS seems to issue orientationchange-related events in a different order to Android so one has to account for this.
 - Too many things end up trying to resize the canvas - especially when sized via CSS absolute positioning.

We studied this for a while, then had decent success by disabling automatic sizing, sizing things ourselves, and then passing the dimensions and pixel ratio to the renderer/effect. I think there should be an option in the config to allow this.